libp2p/NodeTable.cpp

```diff
@@ -270,14 +279,21 @@ vector<shared_ptr<NodeEntry>> NodeTable::nearestNodeEntries(NodeID _target)
     return ret;
 }
 
-void NodeTable::ping(NodeID _toId, NodeIPEndpoint _toEndpoint) const
+void NodeTable::ping(NodeEntry const& _nodeEntry)
 {
-    NodeIPEndpoint src;
-    DEV_GUARDED(x_nodes) { src = m_hostNode.endpoint; }
-    PingNode p(src, _toEndpoint);
-    p.sign(m_secret);
-    LOG(m_logger) << "Sending " << p.typeName() << " to " << _toId << "@" << p.destination;
-    m_socketPointer->send(p);
+    m_timers.schedule(0, [this, _nodeEntry](boost::system::error_code const& _ec) {
+        if (_ec || m_timers.isStopped())
+            return;
+
+        NodeIPEndpoint src;
+        DEV_GUARDED(x_nodes) { src = m_hostNode.endpoint; }
```

Comment:
Why do you have to lock this?

Reply:
Probably I don't anymore, I wanted to go over unnecessary locks later